pkg/p2p/libp2p/connections_test.go

```diff
@@ -308,3 +278,41 @@ func TestConnectWithDisabledQUICAndWSTransports(t *testing.T) {
 	expectPeers(t, s2, overlay1)
 	expectPeersEventually(t, s1, overlay2)
 }
+
+func TestConnectRepeatHandshake(t *testing.T) {
```

Comment:
can you please document what the test is supposed to do? not just this one, it would be nice to have a godoc style comment on all tests

Reply:
Basically, this test is supposed to test if one node sends multiple handshake messages to another node. Because, double handshake can happen in the situation I explained in the PR description, but the try to perform another handshake after will be penalized by disconnect, as stated in the book of swarm. I will add some go doc to this test.